atom/lib/import-js.js

```diff
@@ -0,0 +1,302 @@
+  return editor.getTextInBufferRange(wordRange);
+}
+
+/**
+ * @param {Array<Object>} selectionsToAskFor
+ * @param {Array<Object>} previousSelections
+ */
+function askForSelections(selectionsToAskFor, previousSelections) {
+  const selections = previousSelections || [];
+
+  if (selectionsToAskFor.length === 0) {
+    // We have no more selections to ask for, so we can wrap it up!
+    if (selections.length === 0) {
+      // No selections were chosen.
+      return undefined;
+    }
+
+    // Foo:0,Bar:1
+    const selectionsString = selections
+      .map((selection) => `${selection.word}:${selection.alternativeIndex}`)
+      .join(',');
+
+    /* eslint-disable no-use-before-define */
+    importWord(selectionsString);
+    /* eslint-enable no-use-before-define */
+    return undefined;
+  }
+
+  // Ask for the next selection
+  const selectionToAskFor = selectionsToAskFor.shift();
+
+  // TODO show what word we are asking for
+  const askView = new AskForSelectionView();
+  askView.setItems(selectionToAskFor.alternatives);
+  askView.show();
+  askView.deferred
+    .then((alternativeIndex) => {
+      const selection = {
+        word: selectionToAskFor.word,
+        alternativeIndex,
+      };
+
+      return selections.concat([selection]);
+    })
+    // Selection was cancelled, so let's just move on.
+    .catch(() => selections)
+    .then((newSelections) => {
+      askView.destroy();
+      askForSelections(selectionsToAskFor, newSelections);
+    });
+
+  return undefined;
+}
+
+/**
+ * @param {String} selections
+ */
+function importWord(selections) {
+  const args = [];
+
+  if (selections) {
+    // We have selections to import.
+    args.push('--selections', selections);
+  } else {
+    // There are no selections to import, so let's figure out what the current
+    // word is.
+    const word = getCurrentWord();
+    if (!word) {
+      return;
+    }
+    args.push('--word', word);
+  }
+
+  exec(args)
+    .then((result) => {
+      if (result.ask_for_selections) {
+        askForSelections(result.ask_for_selections, []);
+        return;
+      }
+
+      const editor = atom.workspace.getActiveTextEditor();
+      if (editor.getText() !== result.output) {
+        editor.buffer.setTextViaDiff(result.output);
+      }
+
+      if (result.messages) {
+        atom.notifications.addSuccess(result.messages);
+      }
+    })
+    .catch((error, result) => {
+      const message = result ? result.error.output : error.toString();
+      atom.notifications.addError(message);
+      throw error;
+    });
+}
+
+function gotoWord() {
+  const word = getCurrentWord();
+  if (!word) {
+    return;
+  }
+
+  exec(['--word', word, '--goto'])
+    .then((result) => {
+      const path = `${getCurrentWorkingDirectory()}/${result.output}`;
+      atom.open({ pathsToOpen: [path], newWindow: false });
+
+      if (result.messages) {
+        atom.notifications.addSuccess(result.messages);
```

Comment:
I think this will probably say "ImportJS: imported `foo`" (which is a bug). It's probably good to display messages here though, because we might want to show something in the future.